scripts/upgrade

```diff
@@ -57,186 +70,180 @@ ynh_clean_setup () {
-# CHECK THE PATH
-#=================================================
-
-# Normalize the URL path syntax
-path_url=$(ynh_normalize_url_path $path_url)
-
+# STANDARD UPGRADE STEPS
 #=================================================
-# INSTALL DEPENDENCIES
+# UPGRADE PHP7.3
 #=================================================
 
-ynh_install_php7
+if [ "$(lsb_release --codename --short)" = "buster" ]; then
+	pkg_dependencies="$pkg_dependencies $extra_pkg_dependencies"
+else
+	if [ "$(ynh_app_setting_get --app=$app --key=php_version)" != "7.3" ]; then
+		ynh_print_info --message="Upgrading PHP7.3..."
+		ynh_remove_fpm_config
+		ynh_remove_app_dependencies
+		ynh_remove_php
+		ynh_install_php --phpversion="7.3" --package="$extra_pkg_dependencies"
+	else
+		ynh_install_php --phpversion="7.3" --package="$extra_pkg_dependencies"
+	fi
+fi
 
-#=================================================
-# STANDARD UPGRADE STEPS
 #=================================================
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
 
-# Download, check integrity, uncompress and patch the source from app.src and do required installation for update
-
-# version before upgrade
-old_ver=`sudo yunohost app info $app --output-as plain | ynh_get_plain_key version`
-
-#version check for 0.3
-version=`cd /var/www/$app && bin/console kimai:version --short 2> /dev/null || sed -n 's/^future-release=//p' .github_changelog_generator`
-
-# Uncompress latest sources
-# Create a temporary directory
-tmpdir="$(mktemp -d)"
-# Backup the config file in the temp dir
-if [ -f $final_path/config/packages/local.yaml ]; then
-	cp -af "$final_path/config/packages/local.yaml" "$tmpdir/."
-fi
-if [ -d $final_path/var/invoices ]; then
-	cp -af "$final_path/var/invoices" "$tmpdir/."
-fi
-if [ -f $final_path/var/data/kimai.sqlite ]; then
-	cp -a "$final_path/var/data/kimai.sqlite" "$tmpdir/."
-	ynh_app_setting_set $app database "sqlite"
-fi
-
-
-ynh_setup_source "$final_path"
-
-# If the version is 0.3, symfony update is required
-if [ "$version" = "0.3" ]
+if [ "$upgrade_type" == "UPGRADE_APP" ]
 then
-    ynh_secure_remove "$final_path"
-    ynh_setup_source "$final_path"
-	init_symfony_update $final_path
-fi
-ynh_secure_remove "$final_path"
-ynh_setup_source  "$final_path"
-if [ -f $tmpdir/local.yaml ]; then
-	cp -af "$tmpdir/local.yaml" "$final_path/config/packages/."
-fi
-if [ -d $tmpdir/invoices ]; then
-	cp -af "$tmpdir/invoices" "$final_path/var/."
-fi
-if [ -f $tmpdir/kimai.sqlite ]; then
-sudo cp -af "$tmpdir/kimai.sqlite" "$final_path/var/data/."
+	ynh_print_info --message="Upgrading source files..."
+	
+	# Create a temporary directory and backup config
+	tmpdir="$(mktemp -d)"
+	if [ -f $final_path/config/packages/local.yaml ]; then
+		cp -af "$final_path/config/packages/local.yaml" "$tmpdir/."
+	fi
+	if [ -d $final_path/var/invoices ]; then
+		cp -af "$final_path/var/invoices" "$tmpdir/."
+	fi
+	if [ -f $final_path/var/data/kimai.sqlite ]; then
```

Comment:
Can you please backup the complete `/var/data/` directory. It is used by plugins to store data.

Reply:
Done. all /var/data is now backuped/restored during upgrade